### PR TITLE
 Add test code for system/Cache/Handlers/RedisHandler.php

### PR DESCRIPTION
**Pull request by KazuakiM**
* system/Cache/Handlers/RedisHandler.php
I changed to fix to consider multiple Redis servers.
And we set the response format of ```getMetaData``` same as ``` MemcachedHandler.php```.

* tests/system/Cache/Handlers/RedisHandlerTest.php (new)
I added a test code, but it is not very good.
I think that memory capacity and access speed are not good because all are stored in Hash structure.

### CODE REVIEW

File: system/Cache/Handlers/RedisHandler.php [hunk 1]
```diff
@@ -70,13 +70,13 @@ class RedisHandler implements CacheInterface
...
 	//--------------------------------------------------------------------
 
-	public function __construct($config)
+	public function __construct(array $config)
 	{
-		$this->prefix = $config->prefix ?: '';
+		$this->prefix = $config['prefix'] ?? '';
 
-		if (isset($config->redis))
+		if (!empty($config))
```
Comment (lonnieezell) [inline system/Cache/Handlers/RedisHandler.php]: I know it's nit-picky, but according to our style guide, there should be a space after the exclamation.

File: tests/system/Cache/Handlers/RedisHandlerTest.php [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+	public function testDelete()
+	{
+		$this->redisHandler->save(self::$key1, 'value');
+
+		$this->assertTrue($this->redisHandler->delete(self::$key1));
+		$this->assertFalse($this->redisHandler->delete(self::$dummy));
+	}
+
+	//FIXME: I don't like all Hash logic very much. It's wasting memory.
```
Comment (lonnieezell) [inline tests/system/Cache/Handlers/RedisHandlerTest.php]: Which part are you concerned about here?

File: system/Cache/Handlers/RedisHandler.php (hunk 1 above)
Comment (KazuakiM) [inline system/Cache/Handlers/RedisHandler.php]: OK, I update coding style and previous RP code too.

File: tests/system/Cache/Handlers/RedisHandlerTest.php (hunk 2 above)
Comment (KazuakiM) [inline tests/system/Cache/Handlers/RedisHandlerTest.php]: Redis strcture is following now.  So 'data' field is not exists.
```
'key' : [
    '__ci_type' : xxxxx,
    '__ci_value': yyyyy
]
```
And I think Redis have many functions. I want to use ```set```, ```get``` and etc too.
```set``` is very simple and a small amount of memory.

Comment (lonnieezell) [inline tests/system/Cache/Handlers/RedisHandlerTest.php]: Sounds like you know more than I do here. I'm actually not very familiar with the intricacies of Redis, and have only used it a little.

**Comment by jim-parry** [comment]
not to be too picky, but we expect commits to be GPG-signed :)

**Comment by KazuakiM** [comment]
Hi @jim-parry , @lonnieezell 
I thought that GPG was signed.

**Comment by jim-parry** [comment]
@KazuakiM Your last commit is indeed GPG-signed, thank you :) (the earlier two weren't)

**Comment by lonnieezell** [comment]
@KazuakiM Are you done with this or still working on it?

**Comment by KazuakiM** [comment]
@lloricode 
This PR has been completed.

**Comment by lonnieezell** [comment]
Thanks. Merging in.

**Pull request merged**